control-plane/gateways/dataplane.go

```diff
@@ -0,0 +1,174 @@
+		Name:          "proxy-health",
+		ContainerPort: int32(constants.ProxyDefaultHealthPort),
+	})
+	// Configure the resource requests and limits for the proxy if they are set.
+	if resources != nil {
+		container.Resources = *resources
+	}
+
+	// If running in vanilla K8s, run as root to allow binding to privileged ports;
+	// otherwise, allow the user to be assigned by OpenShift.
+	container.SecurityContext = &corev1.SecurityContext{
+		ReadOnlyRootFilesystem: pointer.Bool(true),
+		// Drop any Linux capabilities you'd get as root other than NET_BIND_SERVICE.
+		Capabilities: &corev1.Capabilities{
+			Add:  []corev1.Capability{netBindCapability},
+			Drop: []corev1.Capability{allCapabilities},
+		},
+	}
+	if !config.EnableOpenShift {
+		container.SecurityContext.RunAsUser = pointer.Int64(0)
+	}
+
+	return container, nil
+}
+
+func getDataplaneArgs(namespace string, config GatewayConfig, bearerTokenFile string, name string) ([]string, error) {
+	proxyIDFileName := "/consul/connect-inject/proxyid"
+	envoyConcurrency := defaultEnvoyProxyConcurrency
+
+	args := []string{
+		"-addresses", config.ConsulConfig.Address,
+		"-grpc-port=" + strconv.Itoa(config.ConsulConfig.GRPCPort),
+		"-proxy-service-id-path=" + proxyIDFileName,
+		"-log-level=" + config.LogLevel,
+		"-log-json=" + strconv.FormatBool(config.LogJSON),
+		"-envoy-concurrency=" + strconv.Itoa(envoyConcurrency),
+	}
+
+	consulNamespace := namespaces.ConsulNamespace(namespace, config.EnableNamespaces, config.ConsulDestinationNamespace, config.EnableNamespaceMirroring, config.NamespaceMirroringPrefix)
+
+	if config.AuthMethod != "" {
+		args = append(args,
+			"-credential-type=login",
+			"-login-auth-method="+config.AuthMethod,
+			"-login-bearer-token-path="+bearerTokenFile,
+			"-login-meta="+fmt.Sprintf("gateway=%s/%s", namespace, name),
+		)
+		if config.ConsulPartition != "" {
+			args = append(args, "-login-partition="+config.ConsulPartition)
+		}
+	}
+	if config.EnableNamespaces {
+		args = append(args, "-service-namespace="+consulNamespace)
+	}
+	if config.ConsulPartition != "" {
+		args = append(args, "-service-partition="+config.ConsulPartition)
+	}
+	if config.TLSEnabled {
+		if config.ConsulTLSServerName != "" {
+			args = append(args, "-tls-server-name="+config.ConsulTLSServerName)
+		}
+		if config.ConsulCACert != "" {
+			args = append(args, "-ca-certs="+constants.LegacyConsulCAFile)
+		}
+	} else {
+		args = append(args, "-tls-disabled")
+	}
+
+	// Configure the readiness port on the dataplane sidecar if proxy health checks are enabled.
+	args = append(args, fmt.Sprintf("%s=%d", "-envoy-ready-bind-port", constants.ProxyDefaultHealthPort))
+
+	args = append(args, fmt.Sprintf("-envoy-admin-bind-port=%d", 19000))
+
+	// Set a default scrape path that can be overwritten by the annotation.
+	prometheusScrapePath := defaultPrometheusScrapePath
+	args = append(args, "-telemetry-prom-scrape-path="+prometheusScrapePath)
```

Comment:
Similarly, I'd be in favor of dropping the metrics-related stuff since we have separate work scoped for all those pieces as well

Reply:
Dropped